application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/DefaultNativeConnectionProvider.java

```diff
@@ -98,31 +100,44 @@ private static LocalNativeConnectionProvider establishConnection(
             final LocalNativeConnectionProvider.Builder builder,
             final String host,
             final int port,
-            final long timeout)
+            final long timeout,
+            final RetryPolicy retryPolicy)
     {
-        long startTime = System.currentTimeMillis();
-        long endTime = startTime + timeout;
-        while (endTime > System.currentTimeMillis())
+        for (int count = 1; count <= retryPolicy.getMaxAttempts(); count++)
```

Comment:
Preference perhaps, but I would rename "count" to "attempt" just to be crystal clear of what it is. ;-)